Automattic/wp-calypso
# Always show Add Email link in `EmailListInactive`

#### Proposed Changes

Letero is iterating on how the email management pages are displayed to users who aren't permitted to add/manage email subscriptions for domains.

This PR covers one of the steps in that process, namely to always show the `Add Email` button in `EmailListInactive`, regardless of whether the current user can add email to the domain in question or not. We will change `EmailProvidersStackedComparison` in #68158 and use that component to communicate the permissions issue to users. **This PR should not be merged before that change is ready.**

Crucially, I also took this opportunity to convert both `EmailListActive` and `EmailListInactive` to Typescript.

| Before | After |
| - | - |
| ![before](https://user-images.githubusercontent.com/1101677/191752775-272e8447-9ddf-40ee-af1b-64a3d035338c.png) | ![after](https://user-images.githubusercontent.com/1101677/191752799-e513172e-3e27-42d3-9377-ff167cef037f.png) |

#### Testing Instructions



1. Have a site with more than one domain, where at least one has no email subscription
2. Add another admin user to that site (`/people/new/:site_slug`)
3. Log in as that other user, accept the invitation
4. Navigate to `/email/:site_slug`
5. Ensure that the `Add Email` button is visible for every domain in `EmailListInactive`

#### Pre-merge Checklist






Related to #
